org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corext/refactoring/code/ExtractFieldRefactoring.java

```diff
@@ -272,6 +271,11 @@ public RefactoringStatus checkInitialConditions(IProgressMonitor pm) throws Core
 			}
 			pm.worked(1);
 
+			if (getMethodDeclaration() == null) {
+				return RefactoringStatus.createFatalErrorStatus("Cannot extract field");

```

Comment:
I know its just being caught by the `finally`, and we're not really providing an API for others, but how about just replacing the string with [`RefactoringCoreMessages.ExtractFieldRefactoring_cannot_extract`](https://github.com/eclipse/eclipse.jdt.ls/blob/master/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corext/refactoring/refactoring.properties#L298) which is pretty close to what we want.

Reply:
Fixed.